core/whistle-1.0.0/src/api/wapi_sms.erl

```diff
@@ -174,43 +294,67 @@ resume_v(JObj) -> resume_v(wh_json:to_proplist(JObj)).
 %%--------------------------------------------------------------------
 -spec bind_q(ne_binary(), wh_proplist()) -> 'ok'.
 bind_q(Queue, Props) ->
-    CallId = props:get_value('call_id', Props, <<"*">>),
-    bind_q(Queue, CallId, props:get_value('restrict_to', Props)).
-
-bind_q(Queue, CallId, 'undefined') ->
-    amqp_util:bind_q_to_exchange(Queue, ?SMS_ROUTING_KEY(CallId), ?SMS_EXCHANGE),
-    amqp_util:bind_q_to_exchange(Queue, ?DELIVERY_ROUTING_KEY(CallId), ?SMS_EXCHANGE),
-    amqp_util:bind_q_to_exchange(Queue, ?RESUME_ROUTING_KEY(CallId), ?SMS_EXCHANGE);
-bind_q(Queue, CallId, ['routing'|Restrict]) ->
-    amqp_util:bind_q_to_exchange(Queue, ?SMS_ROUTING_KEY(CallId), ?SMS_EXCHANGE),
-    bind_q(Queue, CallId, Restrict);
-bind_q(Queue, CallId, ['delivery'|Restrict]) ->
-    amqp_util:bind_q_to_exchange(Queue, ?DELIVERY_ROUTING_KEY(CallId), ?SMS_EXCHANGE),
-    bind_q(Queue, CallId, Restrict);
-bind_q(Queue, CallId, ['resume'|Restrict]) ->
-    amqp_util:bind_q_to_exchange(Queue, ?RESUME_ROUTING_KEY(CallId), ?SMS_EXCHANGE),
-    bind_q(Queue, CallId, Restrict);
-bind_q(_, _, []) -> 'ok'.
+    CallId = props:get_value('call_id', Props, props:get_value('message_id', Props, <<"*">>)),
+    SystemId = ?LOWER(props:get_value('system_id', Props, <<"*">>)),
+    RouteId = ?LOWER(props:get_value('route_id', Props, <<"*">>)),
+    Exchange = props:get_value('exchange', Props, ?SMS_EXCHANGE),
```

Comment:
This is probably where you should declare this custom exchange. Maybe a function to get the exchange:

```erlang
get_exchange(Props) ->
    case props:get_value('exchange', Props) of
        'undefined' -> ?SMS_EXCHANGE;
        ?SMS_EXCHANGE -> ?SMS_EXCHANGE;
        {Exchange, Type} ->
            get_exchange(Exchange, Type, []);
        {Exchange, Type, Options} ->
            get_exchange(Exchange, Type, Options)
        end.

get_exchange(Exchange, Type, Options) ->
    amqp_util:new_exchange(Exchange, Type, Options),
    Exchange.
```

Code is obviously rough-draft.